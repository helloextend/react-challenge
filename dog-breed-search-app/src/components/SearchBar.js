import { useState } from 'react';
import '../index.css'
import {FcSearch} from 'react-icons/fc'


export const SearchBar = ({onSearch}) => {
    const [breed, setBreed] = useState('');
    const handleSearch = () => {
        onSearch(breed);
      };

    return (
        <>
            <div className="seach-div" > 
                <input type="text" className="seach-bar" style={{height:'36px', width:'560px'}} placeholder="Enter Breed Name" value ={breed} onChange={(e)=>setBreed(e.target.value)} />
                <button type="submit" className="seach-btn" style={{width:'105px', height:'40px', backgroundColor:'#0794E3', color:'white', fontSize:'14px', textAlign:'center'}} onClick={handleSearch}> <FcSearch style={{alignContent:'center'}}/> Search</button>
            </div>
            {breed.length === 0 &&(
                <p>Enter the breed please </p>
            )}
        </>

    )
}