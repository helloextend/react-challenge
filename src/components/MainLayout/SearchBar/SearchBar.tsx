import React, { FC } from 'react'
import styled from '@emotion/styled'

import { colors } from '../../../assets/index';
import { Searches, countOccurences } from '../../../assets/custom';

interface Props {
  handleSearch: (event: React.MouseEvent<HTMLFormElement>) => void,
  previousSearches: Searches
}

const SearchBar: FC<Props> = ({ handleSearch, previousSearches }) => {
  const renderSearchButtons = (searches) => {
    const allButtons = [];

    const validCount = countOccurences(searches.valid);
    Object.keys(validCount).forEach(item => {
      const times = validCount[item] === 1 ? '' : ` x${validCount[item]}`;
      allButtons.push(<PreviousSearchButton key={item}>{item + times}</PreviousSearchButton>)
    })

    const invalidCount = countOccurences(searches.invalid);
    Object.keys(invalidCount).forEach(item => {
      const times = invalidCount[item] === 1 ? '' : ` x${invalidCount[item]}`;
      allButtons.push(<PreviousSearchButton style={{ color: colors.vividRed, background: colors.veryPaleRed }} key={item}>{item + times}</PreviousSearchButton>)
    })

    return allButtons;
  }
  return (
    <Form onSubmit={handleSearch}>
      <Input type="text" id="searchTerm" />
      <Button type="submit">Search</Button>
      <PreviousSearchContainer>
        {(previousSearches.valid[0] || previousSearches.invalid[0]) && renderSearchButtons(previousSearches)}
      </PreviousSearchContainer>
    </Form>
  )
}



const Form = styled.form({
  position: 'relative'
})

const Input = styled.input({
  margin: '30px auto 0',
  width: '560px',
  height: '36px',
  background: colors.mostlyWhiteGrey,
  borderRadius: '4px',
  border: 'none',
  borderTopRightRadius: '0',
  borderBottomRightRadius: '0',
  fontFamily: 'Nunito Sans',
  fontStyle: 'normal',
  fontWeight: 'normal',
  fontSize: '16px',
  lineHeight: '22px',
  paddingLeft: '20px',
  paddingRight: '125px',
  ":focus": { outline: "none" },

})

const Button = styled.button({
  position: 'absolute',
  left: '455px',
  top: '30px',
  width: '105px',
  height: '36px',
  fontFamily: 'Nunito Sans',
  fontStyle: 'normal',
  fontWeight: 'bold',
  fontSize: '14px',
  lineHeight: '19px',
  textAlign: 'center',
  color: colors.white,
  background: '#0794E3',
  borderRadius: '4px',
  border: 'none'
})

const PreviousSearchContainer = styled.div({
  marginTop: '10px',
  padding: '0 10px',

})

const PreviousSearchButton = styled.button({
  padding: '0 16px',
  height: '32px',
  lineHeight: '16px',
  fontFamily: 'Nunito Sans',
  color: '#64A364',
  borderRadius: '29px',
  border: 'none',
  background: '#E9F9EA',
  fontStyle: 'normal',
  fontWeight: 'normal',
  fontSize: '12px',
  marginRight: '10px',
  marginBottom: '10px'
})



export default SearchBar
