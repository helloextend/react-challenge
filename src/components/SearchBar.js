import React, { Component } from 'react'
import styled from '@emotion/styled'

export class SearchBar extends Component {
  render() {
    return (
      <Container>
        <input></input>
        <button>Search</button>
      </Container>
    )
  }
}

const Container = styled.div({
  display: 'flex',
  justifyContent: 'space-between',
})

export default SearchBar
