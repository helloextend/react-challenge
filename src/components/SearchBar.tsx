import React, { FC, useState } from "react";
import styled from "@emotion/styled";
import { colors } from "../assets";
import { icons } from "../assets/icons/index";
import { useSelector, useDispatch } from "react-redux";
import { RootState } from "../redux/store";
import { dogBreedHandler } from "../redux/listDogs-slice";

const Searchbar: FC = () => {
  const [dogBreed, setDogBreed] = useState("");

  const inputHandler = (e: React.ChangeEvent<HTMLInputElement>) => {
    setDogBreed(e.target.value);
  };

  const breed = useSelector((state: RootState) => state.dogBreed.breed);
  const dispatch = useDispatch();

  const searchForDogsHandler = () => {
    dispatch(dogBreedHandler(dogBreed));
  };

  return (
    <Container>
      <SearchBar>
        <input
          id="searchBox"
          placeholder="Affenpinscher"
          style={{
            width: "281px",
            marginLeft: "17px",
            fontFamily: "Nunito sans",
            fontSize: "16px",
            letterSpacing: "0px",
            textAlign: "left",
            backgroundColor: colors.mostlyWhiteGrey,
            border: "none",
          }}
          onChange={inputHandler}
        />
        <button
          style={{
            width: "105px",
            height: "36px",
            backgroundColor: colors.blue,
            borderRadius: "4px",
            display: "inline-flex",
            alignItems: "center",
            justifyContent: "space-evenly",
            border: "none",
            cursor: "pointer",
          }}
          onClick={searchForDogsHandler}
        >
          <img
            src={icons.searchIcon}
            alt="search icon"
            style={{ width: "20px", height: "20px" }}
          ></img>
          <span
            style={{
              color: "white",
              fontWeight: 700,
              fontSize: "12px",
              textAlign: "center",
            }}
          >
            Search
          </span>
        </button>
      </SearchBar>
    </Container>
  );
};

const Container = styled.div({
  display: "flex",
  justifyContent: "space-between",
  marginTop: "64px",
});

const SearchBar = styled.div({
  width: "560px",
  height: "36px",
  borderRadius: "4px",
  display: "flex",
  alignItems: "center",
  backgroundColor: colors.mostlyWhiteGrey,
  justifyContent: "space-between",
  overflow: "hidden",
});

export default Searchbar;
