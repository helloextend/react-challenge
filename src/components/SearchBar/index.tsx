import React, { FC, useState, useEffect } from 'react'
import styled from '@emotion/styled'
import axios from 'axios'
import { useDispatch, useSelector } from 'react-redux'
import searchIcon from '../../assets/icons/search-icon.png'
import { fetchDogs } from '../../redux/actions'

const SearchBar: FC = () => {

  const [query, setQuery] = useState<String>("")

  const dispatch = useDispatch()
  const response = useSelector((state) => state.dogsList)

  const handleFormSubmit = async (e: React.FormEvent<HTMLFormElement>) => {
    e.preventDefault();
    // Query API
    if (query) {
      dispatch(fetchDogs(query.trim().toLocaleLowerCase()))
      // Clear Search Field
      setQuery('');
    }
  }

  const handleChange = (e: React.ChangeEvent <HTMLInputElement>) => {
    const {value} = e.target;
    setQuery(value);
  }


  return (
    <div>
      <form onSubmit={handleFormSubmit} >
        <div>
          <SearchField type="text" onChange={handleChange} value={query}></SearchField>
          <SearchButton>
              <SearchIcon src={searchIcon}></SearchIcon>
              Search
              </SearchButton>
        </div>
      </form>
    </div>
  )
}

const SearchIcon = styled.img({

    position:'absolute'
    left:'5px',
    top:'5px',
    fontSize:'30px'
})

const SearchField = styled.input({
  width: '455px',
  height: '36px',
  background: '#F7F7F7',
  borderRadius: '4px'
  // Text Style
  fontFamily: 'Nunito Sans',
  fontStyle: 'normal',
  fontWeight: 'normal',
  fontSize: '16px',
  lineHeight: '22px',
  color: '#44484C'
})

const SearchButton = styled.button({
    position:'relative',
    textAlign: 'right',
    hight: '36px',
    width: '105px',
    backgroundColor: '#0794E3',
    borderRadius: '4px'
    display: 'inlineBlock',
    // Text Style
    fontFamily: 'Nunito Sans',
    fontStyle: 'normal',
    fontWeight: 'bold',
    fontSize: '14px',
    lineHeight: '19px'
    color:'#FFFFFF',
    padding: '5px 20px',
    border: 'none',
    }
})

export default SearchBar
