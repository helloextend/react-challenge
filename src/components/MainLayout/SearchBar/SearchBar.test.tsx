import * as React from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'
import SearchBar from './SearchBar'


const baseProps = {
  handleSearch: jest.fn(),
};

describe('SearchBar component', () => {
  it('renders without crashing', () => {
    const wrapper = shallow(<SearchBar {...baseProps} />)
    expect(wrapper.exists()).toBe(true)
  });

  it('calls handleSearch when submitted', () => {
    const wrapper = shallow(<SearchBar {...baseProps} />)
    wrapper.find('Styled(form)').simulate('submit')
    expect(baseProps.handleSearch).toHaveBeenCalled()
  })

  it('renders the UI as expected', () => {
    const tree = renderer
      .create(<SearchBar {...baseProps} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

})
