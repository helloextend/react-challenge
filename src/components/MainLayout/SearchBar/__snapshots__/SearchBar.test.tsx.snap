// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar component renders the UI as expected 1`] = `
<form
  className="css-79elbk"
  onSubmit={[MockFunction]}
>
  <input
    className="css-1ich0l"
    id="searchTerm"
    type="text"
  />
  <button
    className="css-1blg6he"
    type="submit"
  >
    Search
  </button>
</form>
`;
